src/main/java/edu/wpi/grip/core/Connection.java

```diff
@@ -29,6 +30,10 @@ public Connection(EventBus eventBus, Socket<? extends T> outputSocket, Socket<T>
         checkNotNull(inputSocket);
         checkNotNull(outputSocket);
         checkNotNull(eventBus);
+        checkArgument(!Socket.Direction.INPUT.equals(outputSocket.getDirection()),
+                "outputSocket cannot be an input socket");
+        checkArgument(!Socket.Direction.OUTPUT.equals(inputSocket.getDirection()),
+                "inputSocket cannot be an output socket");
```

Comment:
How can a socket check this?  It doesn't know if it's being added as an input or output socket.